go-controller/pkg/ovn/egressip.go

```diff
@@ -545,25 +549,32 @@ func (oc *Controller) deleteEgressNode(egressNode *kapi.Node) error {
 			}
 		}
 		if needsReassignment {
-			klog.V(5).Infof("EgressIP: %s about to be re-assigned", eIP.Name)
-			if err := oc.deleteEgressIP(&eIP); err != nil {
-				klog.Errorf("EgressIP: %s re-assignmnent error: old egress IP deletion failed, err: %v", eIP.Name, err)
-			}
-			eIP = *(eIP.DeepCopy())
-			eIP.Status = egressipv1.EgressIPStatus{
-				Items: []egressipv1.EgressIPStatusItem{},
-			}
-			if err := oc.addEgressIP(&eIP); err != nil {
-				klog.Errorf("EgressIP: %s re-assignmnent error: new egress IP assignment failed, err: %v", eIP.Name, err)
-			}
-			if err := oc.updateEgressIPWithRetry(&eIP); err != nil {
-				klog.Errorf("EgressIP: %s re-assignmnent error: update of new egress IP failed, err: %v", eIP.Name, err)
+			if err := oc.reassignEgressIP(eIP); err != nil {
+				klog.Errorf("EgressIP: %s re-assignmnent error: %v", eIP.Name, err)
 			}
 		}
 	}
 	return nil
 }
 
+func (oc *Controller) reassignEgressIP(eIP egressipv1.EgressIP) error {
```

Comment:
the caller has a `*egressipv1.EgressIP`, which you dereference to pass to this method, which you then make a new pointer to to pass to other methods... you should just take a pointer here, and don't dereference the deepcopy below either